src/librustdoc/html/render/mod.rs

```diff
@@ -96,6 +96,22 @@ crate struct IndexItem {
 #[derive(Debug)]
 crate struct RenderType {
     name: Option<String>,
+    generics: Option<Vec<Generic>>,
+}
+
+/// A type used for the search index.
+#[derive(Debug)]
+crate struct Generic {
```

Comment:
At this point, I think we can simply drop this type and simply keep the string into `RenderType::generics`.